### ISSUE TITLE
Session corruption

### ISSUE THREAD

**Issue by PulsedMedia-Aleksi** [rakshasa/rtorrent#1019]
rTorrent "launches", you get the view and torrent list by a flash and then crashes with following error:

```
Caught internal_error: SocketFd::close() called on an invalid file descriptor
/usr/local/lib/libtorrent.so.21(_ZN7torrent14internal_error10initializeERKSs+0x1ef) [0x7fada1c7f89f]
rtorrent(_ZN7torrent14internal_errorC2EPKc+0x44) [0x4585d4]
/usr/local/lib/libtorrent.so.21(+0xbccd4) [0x7fada1d02cd4]
/usr/local/lib/libtorrent.so.21(+0x9a4cd) [0x7fada1ce04cd]
/usr/local/lib/libtorrent.so.21(+0x95030) [0x7fada1cdb030]
/usr/local/lib/libtorrent.so.21(_ZN7torrent10DhtManager5startEt+0x69) [0x7fada1c7d5f9]
rtorrent() [0x4b0cde]
rtorrent() [0x4c2de5]
rtorrent() [0x4c81be]
/usr/local/lib/libtorrent.so.21(+0xb835a) [0x7fada1cfe35a]
/usr/local/lib/libtorrent.so.21(+0x35953) [0x7fada1c7b953]
/usr/local/lib/libtorrent.so.21(_ZN7torrent11thread_base10event_loopEPS0_+0x15a) [0x7fada1ccac5a]
rtorrent() [0x413bad]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xfd) [0x7fada04d0ead]
rtorrent() [0x414555]

```

rTorrent version: 0.9.8 release

If session is cleared issue goes away.

This probably needs more error checking and error recovery --> Plus clear error message of which torrent causes this so user can remove the issue causing torrent.

**Comment by PulsedMedia-Aleksi** [comment]
We have put a 100€ bounty towards this issue as well -- double if using as service credit with us.
Rakshasa is not affiliated with this, more info: https://wiki.pulsedmedia.com/index.php/Pulsed_Media#Development_Bounty_Program

**Comment by kannibalox** [comment]
Similar to #1003, this can be triggered when rtorrent runs out of file handles.

**Comment by PulsedMedia-Aleksi** [comment]
Solved #1003 - relation to this is unknown yet (no remaining test case)

**Comment by PulsedMedia-Aleksi** [comment]
Fixed by @kannibalox suggestion on #1003 

@kannibalox can you contact us for payment?
